### PR TITLE
New Login Design

### PR DESCRIPTION
**Pull request by saketkumar95**
- [x] SeverUrl Page
- [ ] Login Page

Landscape Screen
screenshot_20170528-150617

Keyboard Up Screen
screenshot_20170528-150608

New ServerURL Screen
screenshot_20170528-150602

@borisyankov @timabbott @neerajwahi Is this all fine?

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -160,6 +160,7 @@ dependencies {
     compile fileTree(dir: "libs", include: ["*.jar"])
     compile "com.android.support:appcompat-v7:23.0.1"
     compile "com.facebook.react:react-native:+"  // From node_modules
+    compile 'com.facebook.fresco:animated-gif:0.11.0' // For animated GIF support
```
Comment (vishwesh3) [inline android/app/build.gradle]: Why this? Are we using GIF or just png image?

File: src/start/RealmScreen.js [hunk 2]
```diff
@@ -74,21 +75,22 @@ class RealmScreen extends React.Component {
     const { progress, realm, error } = this.state;
 
     return (
-      <Screen title="Welcome" keyboardAvoiding>
+      <Screen title="Welcome to Zulip" keyboardAvoiding>
```
Comment (vishwesh3) [inline src/start/RealmScreen.js]: We would also want to change in `/src/i18n/translations.`

File: src/start/RealmScreen.js [hunk 3]
```diff
@@ -74,21 +75,22 @@ class RealmScreen extends React.Component {
...
         <ScrollView
           ref={(scrollView) => { this.scrollView = scrollView; }}
           centerContent
           keyboardShouldPersistTaps="always"
           onContentSizeChange={() => this.scrollView.scrollToEnd({ animated: true })}
         >
           <View style={[styles.container, moreStyles.container]}>
-            <Label text="Your server URL" />
+            <Image style={[styles.image]} source={require('../../static/img/bus.png')} />
+            <Label text="Server URL" />
```
Comment (vishwesh3) [inline src/start/RealmScreen.js]: Same here (/src/i18n/translations).

File: src/start/RealmScreen.js [hunk 4]
```diff
@@ -74,21 +75,22 @@ class RealmScreen extends React.Component {
...
+      <Screen title="Welcome to Zulip" keyboardAvoiding>
         <ScrollView
           ref={(scrollView) => { this.scrollView = scrollView; }}
           centerContent
           keyboardShouldPersistTaps="always"
           onContentSizeChange={() => this.scrollView.scrollToEnd({ animated: true })}
         >
           <View style={[styles.container, moreStyles.container]}>
-            <Label text="Your server URL" />
+            <Image style={[styles.image]} source={require('../../static/img/bus.png')} />
```
Comment (vishwesh3) [inline src/start/RealmScreen.js]: Here only single style is applied. Better to use `style={styles.image}`

**Comment by vishwesh3** [comment]
Found another open PR https://github.com/zulip/zulip-mobile/pull/511. Is this PR replace that one?

File: android/app/build.gradle (hunk 1 above)
Comment (saketkumar95) [inline android/app/build.gradle]: We need to support gifs too in the future. So added this right now.

File: src/start/RealmScreen.js (hunk 4 above)
Comment (saketkumar95) [inline src/start/RealmScreen.js]: Good catch. Earlier I was using multiples so i added that. :+1:

File: android/app/build.gradle (hunk 1 above)
Comment (vishwesh3) [inline android/app/build.gradle]: ok cool 👍! so we can add this at that time. For now we can remove this, using unwanted dependencies is not good (makes app heavy).

Comment (saketkumar95) [inline android/app/build.gradle]: It will be used in the Login Page which i will have already implemented.

File: src/start/RealmScreen.js (hunk 2 above)
Comment (borisyankov) [inline src/start/RealmScreen.js]: Changing strings adds a lot of overhead to translating.
I don't see the point in this change.

File: src/start/RealmScreen.js [hunk 5]
```diff
@@ -74,21 +75,22 @@ class RealmScreen extends React.Component {
...
+            <Label text="Server URL" />
             <Input
               style={styles.field}
               autoFocus
               autoCorrect={false}
               autoCapitalize="none"
-              placeholder="Server URL"
+              placeholder="https://your-server-url.zulipchat.com"
```
Comment (borisyankov) [inline src/start/RealmScreen.js]: We have discussed before and agreed that having an actual valid value as placeholder to be undesirable.

**Comment by borisyankov** [comment]
@saketkumar95 try to create even more focused PRs.
The more stuff they contain, the more discussions we need to have before merging them.
In this case, I have some concerns about the image, but we keep discussing the text and translations.

**Comment by zulipbot-test** [comment]
Hello @saketkumar95, a Zulip maintainer reviewed this pull request over a week ago, but you haven't updated your pull request since. Please take a look at the requested changes and update your pull request accordingly.

Thank you for your valuable contributions to Zulip!

**Comment by saketkumar95** [comment]
@borisyankov I will open it when we get some nice design for the image. :)

**Pull request closed without merging**